synapse/handlers/federation.py

```diff
@@ -2599,8 +2599,21 @@ def add_display_name_to_third_party_invite(
                 original_invite_id, allow_none=True
             )
         if original_invite:
-            display_name = original_invite.content["display_name"]
-            event_dict["content"]["third_party_invite"]["display_name"] = display_name
+            # If the m.room.third_party_invite event's content is empty, it means the
+            # invite has been revoked.
+            if original_invite.content:
+                display_name = original_invite.content["display_name"]
```

Comment:
shoudn't we handle there not being a display_name property more gracefully?

```suggestion
                display_name = original_invite.content.get("display_name")
```

(which I think makes the condition above redundant anyway)

Reply:
Oh, right, indeed.

Reply:
Fixed in https://github.com/matrix-org/synapse/pull/6147/commits/81d51ce48b449ee55bdcc17b76050283d848d405